woocommerce/woocommerce
# Add totalValue filter

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #42004.
Closes #45172.

This PR adds the `totalValue` filter to wrap the total price on the Cart and Checkout blocks. This filter is needed for plugins such as [Pre-Orders](https://woo.com/products/woocommerce-pre-orders/). It follows the same patterns of the following existing filters:

- `cartItemPrice`
- `subtotalPriceFormat`
- `saleBadgePriceFormat`



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate the [Simple Custom CSS and JS](https://wordpress.org/plugins/custom-css-js/) plugin.
2. Go to `/wp-admin/edit.php?post_type=custom-css-js` and add the following code snippet:

```js
const { registerCheckoutFilters } = window.wc.blocksCheckout;

const modifyTotalsPrice = ( defaultValue, extensions, args, validation ) => {
	return "Pay <price/> now";
};

registerCheckoutFilters( 'my-extension', {
	totalValue: modifyTotalsPrice,
} );
```

> [!IMPORTANT]
> Make sure to run the script in the footer as it requires other dependencies to load first.

3. Go to the frontend and add a product to the cart.
4. Go to the cart page and verify that the total price shows as `Pay ... now`.
5. Go to the checkout page and verify that the total price shows as `Pay ... now`.
6. Remove `<price/>` from the code snippet and repeat steps 4. and 5.
7. Verify that the error message `Error: Returned value must include <price/>, you passed "Pay now"` is visible.

## Screenshots

<table>
<tr>
<td valign="top">Cart block:
<br><br>
<img width="432" alt="Screenshot 2024-02-27 at 20 21 30" src="https://github.com/woocommerce/woocommerce/assets/3323310/e9144950-596b-45b2-8c95-3ae3c2d95263">
</td>
<td valign="top">Checkout block:
<br><br>
<img width="447" alt="Screenshot 2024-02-27 at 20 22 41" src="https://github.com/woocommerce/woocommerce/assets/3323310/6e5395c7-0d46-44d5-b9bd-7f432fb8c3ea">
</td>
</tr>
</table>



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add totalValue filter.

#### Comment 

</details>
